File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{
    //get a text from usser
    string text = get_string("Text: ");

    //count number of letters
    int let = count_letters(text);

    //count number of words
    int wor = count_words(text);

    //count number of sentences
    int sen = count_sentences(text);

    double L = 100 * (double) let / (double) wor;
    double S =  100 * (double) sen / (double) wor;

    double colemanliaau = 0.0588 * L - 0.296 * S - 15.8;


    if (colemanliaau < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (colemanliaau > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(colemanliaau));
    }


}

// count number of letters in the text
int count_letters(string text)

{
    int n = 0;
    int letters = 0;

    // loop that counts number of letters (not other characters) till \0 (NUL)
    while (text[n] != '\0')

        // Count lower case
    {
        if (text[n] >= 'a' && text[n] <= 'z')
        {
            n++;
            letters++;
        }

        // Count capital letter
        else if (text[n] >= 'A' && text[n] <= 'Z')
        {
            n++;
            letters++;
        }

        //else don´t count
        else if (text[n] != '\0')
        {
            n++;
        }

    }
    return letters;
}

// count number of words in the text
int count_words(string text)

{
    int n = 0;
    int words = 0;

    // loop that counts number of letters (not other characters) till \0 (NUL)
    while (text[n] != '\0')

        // Count lower case
    {
        if (text[n] >= 'a' && text[n] <= 'z')
        {
            n++;
        }

        // Count capital letter
        else if (text[n] >= 'A' && text[n] <= 'Z')
        {
            n++;
        }

        // If we have a space he should count the first word
        else if (text[n] == ' ')
        {
            n++;
            words++;
        }

        // If we have a dot we start the loop all over
        else if (text[n] != '\0')
        {
            n++;
        }


    }
    if (n > 0)
    {
        words++;
    }
    return words;

}

// count number of sentences in the text
int count_sentences(string text)

{
    int n = 0;
    int sentences = 0;
    int words = 0;

    // loop that counts number of letters (not other characters) till \0 (NUL)
    while (text[n] != '\0')

        // Count lower case
    {
        if (text[n] == '.' || text[n] == '!' || text[n] == '?')
        {
            n++;
            sentences++;
        }


        // If we have a dot we start the loop all over
        else if (text[n] != '\0')
        {
            n++;
        }

    }
    return sentences;


}

File: readability.py
from cs50 import get_string

def main ():

    # Get a text from usser
    text = get_string("Text: ")

    # Count number of letters
    let = count_letters(text)

    # Count number of words
    wor = count_words(text)

    # Count number of sentences
    sent = count_sentences(text)

    l = 100 * let / wor
    s =  100 * sent / wor

    colemanliaau = 0.0588 * l - 0.296 * s - 15.8


    if colemanliaau < 1:
        print(f"Before Grade 1")
    elif colemanliaau > 16:
        print(f"Grade 16+")
    else:
        print(f"Grade {round(colemanliaau)}")

def count_letters(text):
    letters = 0

    # Loop that counts number of words (not other characters)
    for x in text:
        if x.isalpha():
            letters += 1



    return letters

def count_words(text):
    l = 0
    words = 0
    # Loop that counts number of words
    for x in text:
        if x.isalpha():
            l += 1
        elif x == " ":
            words += 1
    if l > 0:
        words += 1

    return words

def count_sentences(text):
    let = 0
    w = 0
    sentences = 0

    # Loop that counts number of sentences
    for x in text:
        if x in [".","!","?"]:
            sentences += 1

    return sentences
main()